kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerReadTask.java

```diff
@@ -97,7 +97,11 @@ public void doPartialRead() {
         messages = new Vector<>();
       }
 
-      addRecords();
+      Instant now = clock.instant();
+      Duration elapsed = Duration.between(started, now);
+      Duration remaining = requestTimeout.minus(elapsed);
+
+      addRecords(remaining);
```

Comment:
This seems suspicious. It ends up calling `KafkaConsumer.poll(Duration)` and without the check whether the duration is already negative that we do below it looks like this will just cause an `IllegalArgumentException` (see https://kafka.apache.org/30/javadoc/org/apache/kafka/clients/consumer/KafkaConsumer.html#poll(java.time.Duration)). My guess would be that this should not be done and will just HTTP 500 us.